gdk/commands/component/recipe_generator/BuildRecipeGenerator.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import shutil
+from pathlib import Path
+from gdk.common.CaseInsensitive import CaseInsensitiveRecipeFile, CaseInsensitiveDict
+
+import gdk.commands.component.project_utils as project_utils
+import gdk.common.consts as consts
+import gdk.common.utils as utils
+
+
+class BuildRecipeGenerator:
```

Comment:
I think we are using the word "Recipe" too generously. Calling generate which I assume would be a factory method does not actually return anything it just mutates whatever is read from the `component_recipe_file`. There is no encapsulation here and we are mostly mutating  data.

We should be looking a some creational patterns  to organize this the code for example using builder aggregation like this [example](https://refactoring.guru/design-patterns/builder#:~:text=with%20the%20director.-,pseudocode,-This%20example%20of)

Reply:
Also if you think about it. In the context of the GDK, the user is not providing us a recipe the provide us with a RecipeTemplate which we then transform into a template. So maybe CaseInsensitiveRecipeFile is not a good name

Reply:
PS: If we don't want to do this changes now, I would suggest not creating this classes and just making the changes on the methods they were before, later on we could refactor it

Reply:
These classes are created only to handle the responsibility of updating and create the recipe and simplify testing. 
The generator classes create a recipe file in the greengrass-build folders after updating the recipe with the project configuration.

Reply:
I'm not introducing new interfaces or design patterns by only moving the code around for testing.